modules/plugin/coverage-multidim/netcdf/src/main/java/org/geotools/imageio/netcdf/NetCDFGeoreferenceManager.java

```diff
@@ -110,6 +116,13 @@ protected void init() throws FactoryException, IOException {
                 throw new IllegalStateException("Unable to create envelope for this dataset");
             }
 
+            // The previous code was able to automatically convert km coordinates to meter.
+            // Let's check if we still want this automatically conversion occur or we
+            // want the actual axisUnit to be used.
+            if (!NetCDFCRSUtilities.isConvertAxisKm() && axisUnit != null) {
```

Comment:
You said in other parts of the code that the conversion could be due to using feet... so why make it specific to KM?

Reply:
Nope. There wasn't any previous code in the past doing the foot conversion. I suppose is the GT Referencing doing that.